site/content/extend/plugins/webapp/actions.md

```diff
@@ -1,115 +1,493 @@
+This guide assumes you've already set up your plugin development environment for web app plugins to match [mattermost-plugin-starter-template](https://github.com/mattermost/mattermost-plugin-starter-template). If not, follow the README instructions of that repository first, or [see the Hello, World! guide](/extend/plugins/webapp/hello-world/).
 
-## Import mattermost-redux
+## Basic example
 
-First, you'll need to add `mattermost-redux` as a dependency of your web app plugin.
+Here's an [example](https://github.com/mattermost/mattermost-plugin-jira/blob/master/webapp/src/components/modals/create_issue/index.ts) of a web app plugin making use of [mattermost-redux's](https://github.com/mattermost/mattermost-redux) functions:
 
-```bash
-cd /path/to/plugin/webapp
-npm install mattermost-redux
+```javascript
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+
+import {closeCreateModal, createIssue, fetchJiraIssueMetadataForProjects, redirectConnect} from 'actions';
+import {isCreateModalVisible, getCreateModal} from 'selectors';
+
+import CreateIssue from './create_issue_modal';
+
+const mapStateToProps = (state) => {
+    const {postId, description, channelId} = getCreateModal(state);
+    const post = (postId) ? getPost(state, postId) : null;
+    const currentTeam = getCurrentTeam(state);
+
+    return {
+        visible: isCreateModalVisible(state),
+        post,
+        description,
+        channelId,
+        currentTeam,
+    };
+};
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+    close: closeCreateModal,
+    create: createIssue,
+    fetchJiraIssueMetadataForProjects,
+    redirectConnect,
+}, dispatch);
+
+export default connect(mapStateToProps, mapDispatchToProps)(CreateIssue);
 ```
 
-That will add `mattermost-redux` as a dependency in your `package.json` file, allowing it to be imported into any of your plugin's JavaScript files.
+## Some common actions
 
-## Use an Action
+We've listed out some of the commonly-used actions that you can use in your web app plugin. You can find all the actions that are available for your plugin to import [in the source code for mattermost-redux](https://github.com/mattermost/mattermost-redux/tree/master/src/actions).
 
-Actions are used as part of components. To give components access to these actions, we pass them in as React props from the component's container `index.js` file. To demonstrate this, we'll create a new component.
+**createChannel**
 
-In the `webapp` directory, let's create a component folder called `action_example` and switch into it.
+Dispatch this action to create a new channel.
 
-```bash
-mkdir -p src/components/action_example
-cd src/components/action_example
+```javascript
+export function createChannel(channel: Channel, userId: string): ActionFunc {
```

Comment:
Yeah, the signatures really provide a broader idea but as I had suggested above to have the snippets in collapsible divs instead of just repo links, I hope that covers it as well. Probably, someone completely new to these would still want to have a look at its body as well and it's better if we have it here and still not take up that much space on the page. The user can click on whatever they wanna see or else go to the code base to find the additional actions anyways.